tower-buffer/src/lib.rs

```diff
@@ -40,6 +43,54 @@ where
     worker: worker::Handle<T::Error>,
 }
 
+/// Buffer requests with a bounded buffer
+pub struct BufferLayer<E, S, Request> {
```

Comment:
Could `E` be the last generic with a default value of `DefaultExecutor`? `struct BufferLayer<Request, S, E = DefaultExecutor`.